lib/icinga/apiactions.cpp

```diff
@@ -684,18 +680,19 @@ Dictionary::Ptr ApiActions::ExecuteCommand(const ConfigObject::Ptr& object,
 
 			/* Resolve notification macro */
 			String resolved_notification = MacroProcessor::ResolveMacros(
-					notification_string, resolvers, checkable->GetLastCheckResult(), nullptr,
-					MacroProcessor::EscapeCallback(), nullptr, false
+				notification_string, resolvers, checkable->GetLastCheckResult(), nullptr,
+				MacroProcessor::EscapeCallback(), nullptr, false
 			);
 
 			Notification::Ptr notification = Notification::GetByName(resolved_notification);
 			if (!user)
 				return ApiActions::CreateResult(404, "Can't find a valid notification for '" + resolved_notification + "'.");
 
 			/* Get author */
-			String author = "";
-			if (params->Contains("author"))
-				author = HttpUtility::GetLastParameter(params, "author");
+			if (!authenticatedApiUser)
+				return ApiActions::CreateResult(401, "Can't find API user");
+
+			String author = authenticatedApiUser->GetName();
```

Comment:
What's this variable for? You don't use it multiple times and the ='s right hand side isn't that long.